### ISSUE TITLE
Importing types from a js module doesn't work

### ISSUE THREAD

**Issue by zombie** [microsoft/TypeScript#28660]
**TypeScript Version:**  3.2.0-rc

**Code**

Example from [TypeScript 2.9 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#import-types):

```ts
// module.js

class Pet {
  get name() {
    return "";
  }
}

module.exports = {Pet};
```

```ts
// global-script.ts

// error TS2694: Namespace `"module".export=` has no exported member `Pet`.

function adopt(p: import("./module").Pet) {
  console.log(`Adopting ${p.name}...`);
}

// This works, though it's far from ideal, and not always applicable.

let module: typeof import("./module");

function adopt2(p: typeof module.Pet.prototype) {
  console.log(`Adopting ${p.name}...`);
}
```

```js
// a.js

// error TS2694: Namespace `"module".export=` has no exported member `Pet`.

/**
 * @param {import("./module").Pet} p
 */
function walk(p) {
  console.log(`Walking ${p.name}...`);
}

// Works, showing that type information is exported/preserved on some level.

/**
 * @typedef {import("./module")} module
 * @param {typeof module.Pet.prototype} p
 */
function walk2(p) {
  console.log(`Walking ${p.name}...`);
}
```

Typecheck using:

```
tsc -t es2017 --checkjs --allowjs --noemit a.js global-script.ts
```

**Expected behavior:**
Type annotation should work and not error out.

**Actual behavior:**
error TS2694: Namespace `"module".export=` has no exported member `Pet`.

**Related Issues:**
Issue #23936 seems possibly related, but not sure.

**Comment by weswigham** [comment]
I'm not sure if we recognize
```
module.exports = {Pet};
```
As exporting an alias of `Pet` yet.

**Comment by dgoldstein0** [comment]
This doesn't look like a commonjs vs es6 issue to me.

Here's an alternative repro:

```
import * as x from "tslib";

const assign: import("tslib").__assign = x.__assign; // Namespace '"tslib"' has no exported member '__assign'.(2694)
let module: typeof import("tslib");

const assign2: typeof module.__assign = x.__assign;
```
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wB5wGZQiOAIhlQBtgAjQgbgFgAoBgYwgDtV41VgBzVgLjihIsABTEylQgEoAdAH15XXqzgBeTAqWpufOo3qkApvBAQAJgFdjgmAE8wRiNiHhoMcSXJVp+5mw4UHRUAJlsHJxczK2MtZT51TUV41n0gA

this seems to be a general issue with `import()` in the type position, when the target of the import is a declaration file.

**Comment by dgoldstein0** [comment]
since the issue I posted isn't really a JS issue (as this one is tagged) I opened a separate issue for it.  I suppose you can dedupe if you think it's the same issue.

**Comment by jakebailey** [comment]
Looking at old bugs; this was fixed by #41287, which is explicitly to handle the pattern listed in https://github.com/microsoft/TypeScript/issues/28660#issuecomment-441775913
